internal/edgeapi/backend/api.go

```diff
@@ -0,0 +1,65 @@
+package backend
+
+import (
+	"context"
+
+	"github.com/project-flotta/flotta-operator/models"
+)
+
+const (
+	Registered   = RegistrationStatus("registered")
+	Unregistered = RegistrationStatus("unregistered")
+	Unknown      = RegistrationStatus("unknown")
```

Comment:
I decided to not support it at this stage, as it would introduce additional k8s queries for the EDSR and complicate the implementation. If there would be a need to differentiate between unregistered and enrolled, it can be added.